mitodl/open-discussions
# Add channel invitations

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #1400 

#### What's this PR do?

Adds channel invitation functionality

#### How should this be manually tested?
- Go to the contributors tab of a private channel and enter an email address that does not exist as a user.
- You should receive an email that prompts you to register.
- Registering with that email should resolve outstanding invitations into channel memberships

Known Limitations:
- Registering with a different email will not carry over the channel invitation
- Registering still requires an additional email confirmation loop

Email screenshot:
![screenshot_2019-02-22 nathan levesque invited you to evil dinosaurs 2 - rhysyngsun gmail com - gmail](https://user-images.githubusercontent.com/28598/53228070-1473e780-364e-11e9-8e45-9ef55425bc22.png)


#### What GIF best describes this PR or how it makes you feel?
(Optional)
